Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Tooltips/ToolTipSpawner.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections;
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Definitions.InputSystem;
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.ToolTips
+{
+    /// <summary>
+    /// Add to any Object to spawn ToolTips on tap or on focus, according to preference
+    /// Applies its follow settings to the spawned ToolTip's ToolTipConnector component
+    /// </summary>
+    public class ToolTipSpawner : MonoBehaviour, IMixedRealityInputHandler, IMixedRealityFocusHandler
+    {
+        private enum VanishType
```

Comment:
Please add some comments describing the purpose of the enum and the meaning of the values. Will help those reading the code to better understand the usage.

thanks!